dissect/target/loaders/libvirt.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+
+from defusedxml import ElementTree
+
+import dissect.target.container as container
+from dissect.target import Target
+from dissect.target.loader import Loader
+
+
+class LibvirtLoader(Loader):
+    """Load libvirt xml configuration files."""
+
+    def __init__(self, path: Path, **kwargs):
+        path = path.resolve()
+
+        super().__init__(path)
+
+    @staticmethod
+    def detect(path: Path) -> bool:
+        if path.suffix.lower() != ".xml":
+            return False
+
+        with path.open("rb") as fh:
+            return b"<domain>" in fh.read(512)
+
+    def map(self, target: Target) -> None:
+        xml_data = ElementTree.XML(self.path.read_text())
+        for disk in xml_data.findall("devices/disk/source"):
+            if file := disk.get("file"):
+                target_file = self.path.joinpath(file)
```

Comment:
Another question - in a common libvirt setup, is there a relation between where the `.xml` file is and where the virtual disk files are? E.g. are they in the same directory?

I'm asking because sometimes we get a bunch of files in a single directory as "evidence". In that case, we may want to check if a file with the disk name exists in the same directory as the .xml file. The original absolute path is usually only checked as a last resort. This is also done in all similar loaders (`.vmx` etc)